### PR TITLE
executor: remove Next function for StreamAggExec

### PR DESCRIPTION
**Pull request by hsqlu**

### CODE REVIEW

File: executor/aggregate.go [hunk 1]
```diff
@@ -282,50 +282,7 @@ func (e *StreamAggExec) Open(ctx context.Context) error {
 
 // Next implements the Executor Next interface.
 func (e *StreamAggExec) Next(ctx context.Context) (Row, error) {
```
Comment (zz-jason) [inline executor/aggregate.go]: functons `Next` can be totally removed.

**Comment by zz-jason** [comment]
to #5985

File: executor/aggregate.go (hunk 1 above)
Comment (hsqlu) [reply executor/aggregate.go]: Got it

**Comment by zz-jason** [comment]
@hsqlu please fix ci

**Comment by hsqlu** [comment]
PTAL @zz-jason

File: session.go [hunk 2]
```diff
@@ -607,15 +608,17 @@ func createSessionWithDomainFunc(store kv.Storage) func(*domain.Domain) (pools.R
...
-	for {
-		row, err := rs.Next(ctx)
-		if err != nil {
-			return nil, errors.Trace(err)
-		}
-		if row == nil {
-			break
-		}
-		rows = append(rows, row)
+	chk := rs.NewChunk()
```
Comment (zz-jason) [inline session.go]: line 611~621 should be warped with a `for` loop to insure we have got all the results of `rs`

File: session.go [hunk 3]
```diff
@@ -607,15 +608,17 @@ func createSessionWithDomainFunc(store kv.Storage) func(*domain.Domain) (pools.R
...
+	err := rs.NextChunk(ctx, chk)
+	if err != nil {
```
Comment (zz-jason) [inline session.go]: line 613~618 can be simplified as:
```go
if err != nil || chk.NumRows() == 0 {
    return rows, errors.Trace(err)
}
```

File: session.go [hunk 4]
```diff
@@ -607,15 +608,17 @@ func createSessionWithDomainFunc(store kv.Storage) func(*domain.Domain) (pools.R
...
+		return rows, errors.Trace(err)
+	}
+	if chk.NumRows() == 0 {
+		return rows, nil
+	}
+	iter := chunk.NewIterator4Chunk(chk)
+	for r := iter.Begin(); r != iter.End(); r = iter.Next() {
+		rows = append(rows, r)
 	}
 	return rows, nil
```
Comment (hsqlu) [inline session.go]: I saw CI failed. Should I remove this line 623  return rows, nil ?

Comment (zz-jason) [reply session.go]: @hsqlu From CI:
```txt
go get github.com/kisielk/errcheck
go get github.com/golang/lint/golint
vet
session.go:625: unreachable code
Makefile:75: recipe for target 'check' failed
make: *** [check] Error 1
Exited with code 2
```

actually you should remove line 625

Comment (hsqlu) [reply session.go]: Yes, the line number above is unmatched and I fixed it in appending commit.

**Pull request merged**